projects/saturn/src/main/java/io/fairspace/saturn/services/views/ViewService.java

```diff
@@ -70,31 +86,112 @@ public class ViewService {
     private final ViewsConfig viewsConfig;
     private final Dataset ds;
     private final ViewStoreClientFactory viewStoreClientFactory;
+    private final MetadataPermissions metadataPermissions;
+    private final LoadingCache<Boolean, List<FacetDTO>> facetsCache;
+    private final LoadingCache<Boolean, List<ViewDTO>> viewsCache;
 
-    public ViewService(Config.Search searchConfig, ViewsConfig viewsConfig, Dataset ds, ViewStoreClientFactory viewStoreClientFactory) {
+    public ViewService(Config.Search searchConfig,
+                       ViewsConfig viewsConfig,
+                       Dataset ds,
+                       ViewStoreClientFactory viewStoreClientFactory,
+                       MetadataPermissions metadataPermissions) {
         this.searchConfig = searchConfig;
         this.viewsConfig = viewsConfig;
         this.ds = ds;
         this.viewStoreClientFactory = viewStoreClientFactory;
+        this.metadataPermissions = metadataPermissions;
+        this.facetsCache = buildFacetCache();
+        this.viewsCache = buildViewsCache();
+        refreshCaches();
     }
 
-    private Object convertLiteralValue(Object value) {
-        if (value instanceof XSDDateTime) {
-            return ofEpochMilli(((XSDDateTime) value).asCalendar().getTimeInMillis());
-        }
-        return value;
+    public void refreshCaches() {
+        log.info("Caches refreshing/warming up has been triggered");
+        FilteredDatasetGraph.disableQuadPermissionCheck();
+        facetsCache.refresh(Boolean.TRUE);
+        viewsCache.refresh(Boolean.TRUE);
+        FilteredDatasetGraph.enableQuadPermissionCheck();
+        log.info("Caches refreshing/warming up successfully finished");
     }
 
-    @SneakyThrows
-    Range getColumnRange(View view, View.Column column) {
-        if (!EnumSet.of(ColumnType.Date, ColumnType.Number).contains(column.type)) {
-            return null;
+    public List<FacetDTO> getCachedFacets() {
+        try {
+            if (!metadataPermissions.canReadFacets()) {
```

Comment:
Why the check is only for facets, not for cached views too?

Reply:
Fetchiing views is purely static and retrieved from the view config. It has nothing to do with quad retrieving and it was not secured before my changes